src/Text/Pandoc/Writers/Markdown.hs

```diff
@@ -956,23 +957,26 @@ inlineToMarkdown opts lnk@(Link attr txt (src, tit))
                             isEnabled Ext_shortcut_reference_links opts
   ref <- if useRefLinks then getReference attr txt (src, tit) else return []
   reftext <- inlineListToMarkdown opts ref
-  return $ if useAuto
-              then if plain
-                      then text srcSuffix
-                      else "<" <> text srcSuffix <> ">"
-              else if useRefLinks
-                      then let first  = "[" <> linktext <> "]"
-                               second = if txt == ref
-                                           then if useShortcutRefLinks
-                                                   then ""
-                                                   else "[]"
-                                           else "[" <> reftext <> "]"
-                           in  first <> second
-                      else if plain
-                              then linktext
-                              else "[" <> linktext <> "](" <>
-                                   text src <> linktitle <> ")" <>
-                                   linkAttributes opts attr
+  return $
+    let wikilinkString =
+          let deunder c = if c == '_' then ' ' else c
+              deiki s = map deunder $ (reverse . takeWhile (/= '/') . reverse) $ s
```

Comment:
The `deiki` function is also defined in the reader.  If it's really needes, then maybe this can be put in a central place and imported?  Disclaimer: I haven't checked if it's really needed.